### PR TITLE
Handle variable declaration comments

### PR DESCRIPTION
**Pull request by maximderbin**

### CODE REVIEW

File: src/nodes/variable_declaration.js [hunk 1]
```diff
@@ -30,10 +32,28 @@ export function format(node, context, recur) {
...
-        recur(node.declarations[i]);
-        if (node.declarations[i + 1]) {
-            context.write(',\n', context.getIndent());
+        let previous = node.declarations[i - 1];
+        let child = node.declarations[i];
+        let next = node.declarations[i + 1];
+
+        context.write(blockComments.printLeading(child, previous, next));
+
+        if (i > 0) {
```
Comment (maximderbin) [inline src/nodes/variable_declaration.js]: I don't know whyyy I need that!!!???

File: src/nodes/variable_declaration.js [hunk 2]
```diff
@@ -30,10 +32,28 @@ export function format(node, context, recur) {
     }
 
     indent && context.indentIn();
+
     for (let i = 0; i < node.declarations.length; i++) {
-        recur(node.declarations[i]);
-        if (node.declarations[i + 1]) {
-            context.write(',\n', context.getIndent());
+        let previous = node.declarations[i - 1];
+        let child = node.declarations[i];
```
Comment (dmitriiabramov) [inline src/nodes/variable_declaration.js]: s/child/current?

File: src/nodes/variable_declaration.js [hunk 3]
```diff
@@ -30,10 +32,28 @@ export function format(node, context, recur) {
...
+        let next = node.declarations[i + 1];
+
+        context.write(blockComments.printLeading(child, previous, next));
```
Comment (dmitriiabramov) [inline src/nodes/variable_declaration.js]: `printLeading` doesn`t take `next` as an argument

File: src/nodes/variable_declaration.js (hunk 1 above)
Comment (dmitriiabramov) [inline src/nodes/variable_declaration.js]: cause the first one is already indented here https://github.com/dmitriiabramov/esfmt/pull/23/files#diff-5ecfbba207548f6ad3e168d2beddac6cR34

File: src/nodes/variable_declaration.js (hunk 3 above)
Comment (maximderbin) [inline src/nodes/variable_declaration.js]: you was drunk https://github.com/dmitriiabramov/esfmt/blob/master/src/nodes/object_expression.js#L37 but I feel like a shit to think my own head

File: src/nodes/variable_declaration.js (hunk 2 above)
Comment (maximderbin) [inline src/nodes/variable_declaration.js]: just copy&past. Will rename in both places

File: src/nodes/variable_declaration.js (hunk 1 above)
Comment (maximderbin) [inline src/nodes/variable_declaration.js]: argh!!!

File: src/nodes/variable_declaration.js (hunk 3 above)
Comment (maximderbin) [inline src/nodes/variable_declaration.js]: Will fix in both

File: src/nodes/variable_declaration.js [hunk 4]
```diff
@@ -30,10 +32,27 @@ export function format(node, context, recur) {
...
-        recur(node.declarations[i]);
-        if (node.declarations[i + 1]) {
-            context.write(',\n', context.getIndent());
+        let previous = node.declarations[i - 1];
+        let current = node.declarations[i];
+        let next = node.declarations[i + 1];
+
+        context.write(blockComments.printLeading(current, previous));
+
+        if (i > 0) {
```
Comment (maximderbin) [inline src/nodes/variable_declaration.js]: That still smells

**Pull request merged**